packages/analytics-js/src/components/userSessionManager/UserSessionManager.ts

```diff
@@ -510,7 +509,11 @@ class UserSessionManager implements IUserSessionManager {
       state.session.authToken.value = defaultUserSessionValues.authToken;
 
       if (resetAnonymousId) {
-        state.session.anonymousId.value = defaultUserSessionValues.anonymousId;
+        if (this.isPersistenceEnabledForStorageEntry('anonymousId')) {
+          this.setAnonymousId();
+        } else {
+          state.session.anonymousId.value = defaultUserSessionValues.anonymousId;
+        }
```

Comment:
Isn't just this enough? If persistence were not enabled for that field, the value in the state would anyway be the default value.

```suggestion
        // This will generate a new anonymous ID
        this.setAnonymousId();
```